### DISCUSSION TITLE
How to use Multipath Payments

### DISCUSSION THREAD

**Discussion by 17621859515** [lightningnetwork/lnd#7294] [General]
May I ask how to use rest api for Multipath Payments?
Researched for a long time, without success.

**Comment by guggero** [comment]
You can set the `max_parts` to a number bigger than 1 in https://lightning.engineering/api-docs/api/lnd/router/send-payment-v2.

**Reply by 17621859515** [reply]
> You can set the `max_parts` to a number bigger than 1 in https://lightning.engineering/api-docs/api/lnd/router/send-payment-v2.

In fact, the default value of this parameter is 16.
But I still can only pay with the balance of one channel.
I currently have multiple channels, and the balance of the two largest channels is, 10553, 6000
When I want to pay 12000, the transaction will fail and report an error.

Payment status: FAILED, reason: FAILURE_REASON_NO_ROUTE

How can I make a one-time payment of 12000?
![image](https://user-images.githubusercontent.com/69946784/211231256-dfbbe620-06ac-4327-bf40-bf2d3b2f6311.png)

**Reply by guggero** [reply]
Hmm, I'm not sure if with normal multipath payments (MPP) you can send more than the max balance you have in a single channel with the current algorithm. It sounds like it's trying to send everything without splitting first and then doesn't find a route (obviously) but then doesn't try again...
Can you try with atomic multipath payments (AMP) instead? https://docs.lightning.engineering/lightning-network-tools/lnd/amp

**Reply by bitromortac** [reply]
I think this is due to the min shard size which is 10000 sats (unfortunately not configurable), so splitting from 12000 sat to 6000 sat is not allowed. 
https://github.com/lightningnetwork/lnd/blob/ef8cf73d4ce4b485d47ded5fa219992574c928c1/routing/payment_session.go#L72

**Reply by 17621859515** [reply]
> I think this is due to the min shard size which is 10000 sats (unfortunately not configurable), so splitting from 12000 sat to 6000 sat is not allowed.

I also tried to enable this parameter, but I still can't make multi-channel payment.

**Reply by 17621859515** [reply]
> I think this is due to the min shard size which is 10000 sats (unfortunately not configurable), so splitting from 12000 sat to 6000 sat is not allowed.
> 
> https://github.com/lightningnetwork/lnd/blob/ef8cf73d4ce4b485d47ded5fa219992574c928c1/routing/payment_session.go#L72


I'm not trying to split the channel balance, but to do multi-channel balance co-pay for one check.

I tried various combinations of multi-channel balances to no avail.

**Reply by bitromortac** [reply]
What probably happens is that a payment of the whole amount is tried (12000 sat) first, but that's not possible because of the channel balances. The algorithm then splits the amount into half and tries to find a route for those half-parts, but the split amounts (6000 sat) are below the `DefaultShardMinAmt` (10000 sat) and the payment fails. But to be sure I'll try to reproduce your scenario and will let you know. `payinvoice` only has a `--max_shard_size_msat` but no `--min_shard_size_msat`, so the 10000 sat value can't be influenced.

**Reply by 17621859515** [reply]
> payinvoice

![27b21404-fe3e-4fc6-875b-0fd81d64953b](https://user-images.githubusercontent.com/69946784/211995250-6580248a-77cb-48db-8f65-94d575db33ec.jpeg)

**Reply by 17621859515** [reply]
> What probably happens is that a payment of the whole amount is tried (12000 sat) first, but that's not possible because of the channel balances. The algorithm then splits the amount into half and tries to find a route for those half-parts, but the split amounts (6000 sat) are below the `DefaultShardMinAmt` (10000 sat) and the payment fails. But to be sure I'll try to reproduce your scenario and will let you know. `payinvoice` only has a `--max_shard_size_msat` but no `--min_shard_size_msat`, so the 10000 sat value can't be influenced.


Maybe my problem table described something unclear, I'll try to describe it again.

I currently have multiple channels, and the balance of the two largest channels is, 10553, 6000
When I want to pay 12000, the transaction will fail and report an error.

Payment status: FAILED, reason: FAILURE_REASON_NO_ROUTE

How can I make a one-time payment of 12000 using the balance in these two channels

**Reply by bitromortac** [reply]
You explained the problem very well. In principle you should be able to send this payment, but currently it's not directly possible because of some technical reasons as mentioned above, unfortunately.

One workaround is to rebalance some of the 6000 sat into the channel with 10553 sat to have a higher balance than 12000 sat. This way you should be able to send the payment via a single channel. Are these channels with different peers? This only works reliably if the channels are with different peers.

You can try a self-payment by sending yourself 4000 sat to yourself into the 10553 sat channel - this incurs a bit of routing fees. To prepare, you need to look up the channel id of the channel that has 6000 sat. Another step is to find the node pubkey of the peer with the 10553 sat channel.

Then you send the self-payment:
* `lncli addinvoice 4000`
* `lncli payinvoice --outgoing_chan_id <channel_id_6000sat_channel> --last_hop value <remote_pubkey_10553sat_channel> --allow_self_payment <payment_request>`

Afterwards the payment should be possible.

**Reply by 17621859515** [reply]
> You explained the problem very well. In principle you should be able to send this payment, but currently it's not directly possible because of some technical reasons as mentioned above, unfortunately.
> 
> One workaround is to rebalance some of the 6000 sat into the channel with 10553 sat to have a higher balance than 12000 sat. This way you should be able to send the payment via a single channel. Are these channels with different peers? This only works reliably if the channels are with different peers.
> 
> You can try a self-payment by sending yourself 4000 sat to yourself into the 10553 sat channel - this incurs a bit of routing fees. To prepare, you need to look up the channel id of the channel that has 6000 sat. Another step is to find the node pubkey of the peer with the 10553 sat channel.
> 
> Then you send the self-payment:
> 
> * `lncli addinvoice 4000`
> * `lncli payinvoice --outgoing_chan_id <channel_id_6000sat_channel> --last_hop value <remote_pubkey_10553sat_channel> --allow_self_payment <payment_request>`
> 
> Afterwards the payment should be possible.

Thanks, I see.
The reason is that the minimum payment for each channel is not less than 10000 sat
